helpdesk_mgmt_project/models/helpdesk_ticket.py

```diff
@@ -14,6 +16,33 @@ class HelpdeskTicket(models.Model):
         store=True,
     )
 
+    @api.model
+    def _default_company_id(self):
+        if self._context.get("default_project_id"):
+            return (
+                self.env["project.project"]
+                .browse(self._context["default_project_id"])
+                .company_id
+            )
+        return self.env.company
+
+    # Override field
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        compute="_compute_company_id",
+        store=True,
+        readonly=False,
+        required=True,
+        copy=True,
+        default=_default_company_id,
+    )
+
+    @api.depends("project_id.company_id")
+    def _compute_company_id(self):
+        for ticket in self.filtered(lambda ticket: ticket.project_id):
```

Comment:
AFICS, filtering here can leads you to loop on less records than available in self.
IMHO it doesn't respect https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#computed-fields where it is supposed to browse *all* records.
If I'm wrong sorry for buzzing here.
Regards

Reply:
Fixed.